lib/spack/spack/binary_distribution.py

```diff
@@ -935,10 +935,19 @@ def build_tarball(spec, outdir, force=False, rel=False, unsigned=False,
     spackfile_path = os.path.join(
         cache_prefix, tarball_path_name(spec, '.spack'))
 
+    outdir = mirror.push_url
     remote_spackfile_path = url_util.join(
+        mirror.fetch_url, os.path.relpath(spackfile_path, tmpdir))
+    local_spackfile_path = url_util.join(
```

Comment:
I might be getting confused by the variable names, why is this called `local_spackfile_path`?  I might not be the only one, it seems the "remote" url is coming from the mirrors `fetch_url` while the "local" one is coming from the `push_url`. 

@Paul-Ferrell  You aren't thinking of these as separate mirrors, are you?  The idea with allowing mirrors to have separate fetch and push urls is to handle situations where you might want to have your  S3 mirror behind an http proxy and you couldn't push things to a brainless http mirror.

Reply:
I changed those variable names to make things more clear (push_ and fetch_ vs local_ and remote_). 

I've made another slight change. Now if the fetch url exists 'force' will simply ignore it rather than trying to delete it, which in many cases Spack can't do anyway.  So now no 'writing to'/'deleting from' the fetch url.   

I'm not thinking of these as separate mirrors, but as the same mirror connected by an external mechanism or with different paths for security purposes. In my case, the mirror is just a simple web server; we can't use S3. Like the S3 case, pushing is a separate path with separate security mechanisms. Unlike the S3 case, the push to the file server is done with a separate process over client cert authenticated HTTP PUT requests. I guess I could add support for that in Spack, but it seems odd to me.